phovea.js

```diff
@@ -5,28 +5,42 @@
  **************************************************************************** */
 
 //register all extensions in the registry following the given pattern
-module.exports = function(registry) {
+module.exports = function (registry) {
   //registry.push('extension-type', 'extension-id', function() { return System.import('./src/extension_impl'); }, {});
   /// #if include('datatype', 'matrix')
-  registry.push('datatype', 'matrix', function() { return System.import('./src/matrix/Matrix'); }, {});
+  registry.push('datatype', 'matrix', function () {return System.import('./src/matrix/Matrix');}, {});
   /// #endif
   /// #if include('datatype', 'table')
-  registry.push('datatype', 'table', function() { return System.import('./src/table/Table'); }, {});
+  registry.push('datatype', 'table', function () {return System.import('./src/table/Table');}, {});
   /// #endif
   /// #if include('datatype', 'vector')
-  registry.push('datatype', 'vector', function() { return System.import('./src/vector/Vector'); }, {});
+  registry.push('datatype', 'vector', function () {return System.import('./src/vector/Vector');}, {});
   /// #endif
   /// #if include('datatype', 'stratification')
-  registry.push('datatype', 'stratification', function() { return System.import('./src/stratification/Stratification'); }, {});
+  registry.push('datatype', 'stratification', function () {return System.import('./src/stratification/Stratification');}, {});
   /// #endif
   /// #if include('datatype', 'graph')
-  registry.push('datatype', 'graph', function() { return System.import('./src/graph/GraphProxy'); }, {});
+  registry.push('datatype', 'graph', function () {return System.import('./src/graph/GraphProxy');}, {});
   /// #endif
   /// #if include('datatype', 'atom')
-  registry.push('datatype', 'atom', function() { return System.import('./src/atom/Atom'); }, {});
+  registry.push('datatype', 'atom', function () {return System.import('./src/atom/Atom');}, {});
   /// #endif
   /// #if include('tabSyncer', 'selection')
-  registry.push('tabSyncer', 'selection', function() { return System.import('./src/sync/SelectionSyncer'); }, {});
+  registry.push('tabSyncer', 'selection', function () {return System.import('./src/sync/SelectionSyncer');}, {});
   /// #endif
+
+  registry.push('epPhoveaCoreLocale', 'phoveaCoreLocaleEN', function () {
+    return System.import('./src/assets/locales/en/phovea.json').then(function (json) {
+      return {
+        create: function () {
+          return json;
+        }
+      };
+    });
+  }, {
```

Comment:
I would simplify the resource loading as following:

```js
import {resourceLoader} from './src/plugin';

// ...

registry.push('epPhoveaCoreLocale', 'phoveaCoreLocaleEN', function () { return resourceLoader('./src/assets/locales/en/phovea.json'); }, {
    order: 0,
    ns: 'phovea',
    lng: 'en'
  });
```

In plugin.ts

```ts
/**
 * Helper function to create a loader for resource files (e.g., JSON).
 * The imported resource file is returned as it is.
 *
 * @param url Relative URL to the resource file
 */
export function resourceLoader(url: string): Promise<any> {
  return System.import(url).then((data) => {
    return {
      create: () => data
    };
  });
}
```

I'm not sure though, if WebPack really bundles the locales and if they are available. Please test it.

Reply:
Once we found a final solution please update the summary in the first comment and the Wiki.

Reply:
System.import can only work with a hard coded path you cant really pass it a variable.
https://github.com/webpack/webpack/issues/2401
you would need to use some other library like SystemJS to implemented the proposed change.